woocommerce/woocommerce
# Track Linked Products feature events

### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.



### Changes proposed in this Pull Request:





Closes #43801



### How to test the changes in this Pull Request:



Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

1. Make sure feature `New product editor` is enabled under `/wp-admin/admin.php?page=wc-settings&tab=advanced&section=features`
2. Make sure to enable `product-linked` feature flag under `/wp-admin/tools.php?page=woocommerce-admin-test-helper`. WCA Test Helper plugin is required.
3. Make sure to run the tool `Enable wc-admin Tracking` from `/wp-admin/tools.php?page=woocommerce-admin-test-helper` tab `Tools` and change your console level to verbose.
4. Go to `Products` -> `Add new` from the left side menu.
5. Under the `General` tab click `Change product type` link 
<img width="360" alt="image" src="https://github.com/woocommerce/woocommerce/assets/13334210/5c7c36be-6e91-46d9-aaa0-e679b42667df">

6. In the console from the browser's DevTools the `wcadmin_product_template_selector_open` event should be shown
<img width="882" alt="image" src="https://github.com/woocommerce/woocommerce/assets/13334210/709c5f62-ea08-4d46-a5ef-09a103860fc5">

7. Clicking a different product type from the list should trigger 2 events
<img width="750" alt="image" src="https://github.com/woocommerce/woocommerce/assets/13334210/af426724-3f65-4de5-a9e2-07c4239a9fd7">

one when select it `wcadmin_product_template_selector_selected`
<img width="873" alt="image" src="https://github.com/woocommerce/woocommerce/assets/13334210/4569b49e-f392-4f01-8924-7568e9abaa6e">

the other when the type was effectively changed `wcadmin_product_template_changed` 
<img width="870" alt="image" src="https://github.com/woocommerce/woocommerce/assets/13334210/99ed5bc7-d746-41de-8dfc-083d9a24ce59">

8. Then go to the `Linked products` tab
9. When clicking `Choose products for me` from the `Upsells` section
<img width="692" alt="image" src="https://github.com/woocommerce/woocommerce/assets/13334210/6fefb669-ad30-4472-86cb-0afb1b0d9526">

the `wcadmin_linked_products_choose_related_click` event should be shown 
<img width="875" alt="image" src="https://github.com/woocommerce/woocommerce/assets/13334210/5ccbcbaf-add9-4fed-9951-f07677d08e78">

10. When clicking `Choose products for me` from the `Cross-sells` section, the `wcadmin_linked_products_choose_related_click` event should be shown 
<img width="879" alt="image" src="https://github.com/woocommerce/woocommerce/assets/13334210/904516c4-c37b-4b4a-9211-ed2ea2fdc2b8">

11. From the list clicking the product name link 
<img width="704" alt="image" src="https://github.com/woocommerce/woocommerce/assets/13334210/7f1b96cd-264c-4a81-8197-2d4cb3e2d53f">
 should trigger the `wcadmin_linked_products_product_select` event 
<img width="864" alt="image" src="https://github.com/woocommerce/woocommerce/assets/13334210/a6b8151a-403b-46fb-8dd5-8d0656ee8dce">

12. When clicking the preview of a product 
<img width="681" alt="image" src="https://github.com/woocommerce/woocommerce/assets/13334210/fd5acf36-544c-46c1-9580-6e0cfdc44b08">
the `wcadmin_linked_products_product_preview_click` event should be shown 
<img width="870" alt="image" src="https://github.com/woocommerce/woocommerce/assets/13334210/866ca395-7400-4ce8-95c2-4b7d47dd63db">

13. When clicking remove product from the list 
<img width="683" alt="image" src="https://github.com/woocommerce/woocommerce/assets/13334210/398556eb-dcc6-4f8f-80c8-d559fd2523eb">
the `wcadmin_linked_products_product_remove` event should be shown 
<img width="872" alt="image" src="https://github.com/woocommerce/woocommerce/assets/13334210/ffc0a0e8-eca4-4935-9b93-69f4006d3424">

14. When the product list is empty a placeholder should be shown and when dismiss it the `linked_products_placeholder_dismiss` event should be shown.

15. When the a new product is selected from the select control menu and added to the product list the `linked_products_product_add` should be tracked.



### Changelog entry




<details>

#### Significance




#### Type




#### Message 

#### Comment 

</details>
